api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceServiceImpl.java

```diff
@@ -394,8 +399,16 @@ public Workspace saveAndCloneCohortsAndConceptSets(Workspace from, Workspace to)
   @Override
   public WorkspaceAccessLevel getWorkspaceAccessLevel(
       String workspaceNamespace, String workspaceId) {
-    String userAccess =
-        fireCloudService.getWorkspace(workspaceNamespace, workspaceId).getAccessLevel();
+    WorkspaceACL workspaceACL = fireCloudService.getWorkspaceAcl(workspaceNamespace, workspaceId);
+    WorkspaceAccessEntry workspaceAccessEntry =
+        Optional.of(workspaceACL.getAcl().get(userProvider.get().getEmail()))
```

Comment:
If the user has no email (or it's invalid), it would be good to know that as well. If that's impossible, then this is probably fine.

Reply:
That should be covered earlier on in the process. I think handling email validation should be handled by createUser. (All users must have an email, it is how they log in.)